### DISCUSSION TITLE
Is it possible to update document in server without editor?

### DISCUSSION THREAD

**Discussion by dkrym** [ueberdosis/hocuspocus#142] [Q&A]
Hi guys,
Is it possible to somehow update document (or specifically node attribute) without editor and not break the collab document?

Example:  I have a document with taskList and would like to create REST endpoint which I can call with UUID and it would tick the specific taskItem.

Is there any good way to do this or the only way to do it is same way as in frontend (create doc, connect to server as user, sync document, change the taskItem node attribute, disconnect)?

**Comment by bhwstevens3** [comment]
I have the same issue! I have not found a good solution yet!
The only thing I have been able to achieve so far is to connect directly to RocksDB for creating new documents. But for updates and modifications to the existing document, looks like there is no other way than using the provider.
And the problem with the provider is that not only it is not a clear solution, but also I can not fidn an easy way to use it in a transactional way.
Related issues:
https://github.com/ueberdosis/hocuspocus/issues/231
https://github.com/ueberdosis/hocuspocus/issues/222

**Comment by carsonbird** [comment]
This isn't necessarily what you're looking for, but the idea of the tip tap editor is that it's headless, so you could just create an editor on its own on the server, which doesn't require a UI, and you can use all the nice tip tap commands to change the document. Are you not wanting an editor because you think it needs UI? Or for some other reason?
